### PR TITLE
Allow custom cors validation logic

### PR DESCRIPTION
**Pull request by kyelewis**

### CODE REVIEW

File: src/packages/apollo/src/plugins/cors.ts [hunk 1]
```diff
@@ -1,7 +1,22 @@
 import { ApolloServerPlugin } from '@apollo/server';
 import { logger } from '@exogee/logger';
 
-export const Cors: ApolloServerPlugin = {
+// @todo does anyone need this legacy code?
+const defaultValidateOrigin = (origin: string) => {
+	const rootDomain = process.env.ROOT_DOMAIN;
```
Comment (skeep-exogee) [inline src/packages/apollo/src/plugins/cors.ts:6]: I dont think we use this, shall we remove it?

Comment (skeep-exogee) [reply src/packages/apollo/src/plugins/cors.ts:6]: im not sure what `easy-` is

Comment (kyelewis) [reply src/packages/apollo/src/plugins/cors.ts:6]: happy to remove it, it's just the logic in the existing code

Comment (skeep-exogee) [reply src/packages/apollo/src/plugins/cors.ts:6]: yeah i think lets tidy it up we can always add back if its needed

File: src/packages/apollo/src/plugins/cors.ts [hunk 2]
```diff
@@ -1,7 +1,13 @@
 import { ApolloServerPlugin } from '@apollo/server';
 import { logger } from '@exogee/logger';
 
-export const Cors: ApolloServerPlugin = {
+export interface CorsPluginOptions {
+	validateOrigin?(origin: string): boolean;
+}
+
+export const corsPlugin = ({
+	validateOrigin = () => false,
```
Comment (kyelewis) [inline src/packages/apollo/src/plugins/cors.ts:9]: @skeep-exogee do you think the default should be to allow or deny?

Comment (skeep-exogee) [reply src/packages/apollo/src/plugins/cors.ts:9]: yeah as it forces us to supply it 👍

Comment (kyelewis) [reply src/packages/apollo/src/plugins/cors.ts:9]: great, then validate = false is the default we want

**Review by skeep-exogee — APPROVED** [review]

**Pull request merged**